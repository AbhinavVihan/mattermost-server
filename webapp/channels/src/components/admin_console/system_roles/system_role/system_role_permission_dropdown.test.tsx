// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {readAccess} from './types';

import SystemRolePermissionDropdown from './system_role_permission_dropdown';
import {renderWithIntl} from 'tests/react_testing_utils';

describe('admin_console/system_role_permission_dropdown', () => {
    const props = {
        section: {
            name: 'environemnt',
            hasDescription: true,
            subsections: [],
        },
        access: readAccess,
        updatePermissions: jest.fn(),
        isDisabled: false,
    };

    test('should match snapshot', () => {
        const wrapper = renderWithIntl(
            <SystemRolePermissionDropdown
                {...props}
            />);

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot with isDisabledTrue', () => {
        const wrapper = renderWithIntl(
            <SystemRolePermissionDropdown
                {...props}
                isDisabled={true}
            />);

        expect(wrapper).toMatchSnapshot();
    });
});
