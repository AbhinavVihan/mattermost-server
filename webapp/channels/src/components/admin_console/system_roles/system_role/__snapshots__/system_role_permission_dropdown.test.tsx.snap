// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`admin_console/system_role_permission_dropdown should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MenuWrapper "
      >
        <button
          aria-expanded="true"
          class="PermissionSectionDropdownButton dropdown-toggle theme"
          id="systemRolePermissionDropdownenvironemnt"
          type="button"
        >
          <div
            class="PermissionSectionDropdownButton_text"
          >
            Read only
          </div>
          <div
            class="PermissionSectionDropdownButton_icon"
          >
            <i
              class="fa fa-angle-down"
              title="Dropdown Icon"
            />
          </div>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MenuWrapper "
    >
      <button
        aria-expanded="true"
        class="PermissionSectionDropdownButton dropdown-toggle theme"
        id="systemRolePermissionDropdownenvironemnt"
        type="button"
      >
        <div
          class="PermissionSectionDropdownButton_text"
        >
          Read only
        </div>
        <div
          class="PermissionSectionDropdownButton_icon"
        >
          <i
            class="fa fa-angle-down"
            title="Dropdown Icon"
          />
        </div>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`admin_console/system_role_permission_dropdown should match snapshot with isDisabledTrue 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MenuWrapper "
        disabled=""
      >
        <button
          aria-expanded="true"
          class="PermissionSectionDropdownButton dropdown-toggle theme"
          id="systemRolePermissionDropdownenvironemnt"
          type="button"
        >
          <div
            class="PermissionSectionDropdownButton_text"
          >
            Read only
          </div>
          <div
            class="PermissionSectionDropdownButton_icon"
          >
            <i
              class="fa fa-angle-down"
              title="Dropdown Icon"
            />
          </div>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MenuWrapper "
      disabled=""
    >
      <button
        aria-expanded="true"
        class="PermissionSectionDropdownButton dropdown-toggle theme"
        id="systemRolePermissionDropdownenvironemnt"
        type="button"
      >
        <div
          class="PermissionSectionDropdownButton_text"
        >
          Read only
        </div>
        <div
          class="PermissionSectionDropdownButton_icon"
        >
          <i
            class="fa fa-angle-down"
            title="Dropdown Icon"
          />
        </div>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
